astropy/io/ascii/src/tokenizer.c

```diff
@@ -595,7 +596,7 @@ long str_to_long(tokenizer_t *self, char *str)
     char *tmp;
     long ret;
     errno = 0;
-    ret = strtol(str, &tmp, 0);
+    ret = strtol(str, &tmp, 10);
```

Comment:
Yes, see the comment on that commit 7490bdd in the original PR. With the base set to auto-detect, `strtol` would interpret e.g. `0432437` as an `octal` no. and return it as (base-10) `int` `144671`, whereas `0432438` would be identified as invalid octal and promoted to `float`, or with the updated fall-back scheme, to `str`. Hard to imagine anyone has used this 'feature', and anyway it was inconsistent with the Python parser's behaviour.